### PR TITLE
Remove cachemgr.cgi tool

### PR DESCRIPTION
**Pull request by yadij**

### CODE REVIEW

**Review by kinkie — DISMISSED** [review]
big cleanup!

**Review by rousskov — DISMISSED** [review]
Thank you very much for removing this tool!

If this change should be reflected in doc/release-notes/release-7.sgml.in, please adjust accordingly. FWIW, recent commit 80cba6d did adjust those release notes when removing squidclient.

If possible, please adjust the following three leftovers as well:

```
INSTALL:27:the tools/cachemgr.cgi program into your httpd server's cgi-bin
src/cf.data.pre:7533:   SNMP responses, cachemgr.cgi output, squidclient User-Agent request header
tools/purge/purge.1:288:.if !'po4a'hide' .BR cachemgr.cgi "(8)"
```

**Comment by yadij** [comment]
> Thank you very much for removing this tool!
> 
> If this change should be reflected in doc/release-notes/release-7.sgml.in, please adjust accordingly. FWIW, recent commit [80cba6d](https://github.com/squid-cache/squid/commit/80cba6d3a8cb3fa952ed59bc1d3889076d3263f6) did adjust those release notes when removing squidclient.
> 

Done in 64f2af06a96949d8cf365ef6f9768e4f74630873

> If possible, please adjust the following three leftovers as well:
> 
> ```
> INSTALL:27:the tools/cachemgr.cgi program into your httpd server's cgi-bin
> src/cf.data.pre:7533:   SNMP responses, cachemgr.cgi output, squidclient User-Agent request header
> ```

Done in 90bc05c99b08021828da6ad226b6d36710598ca3


> ```
> tools/purge/purge.1:288:.if !'po4a'hide' .BR cachemgr.cgi "(8)"
> ```

This one is left intentionally to avoid conflicts with PR #1541

File: doc/release-notes/release-7.sgml.in
Comment (rousskov) [inline doc/release-notes/release-7.sgml.in:49]: If we do not create an executable with CGI in all-caps:

```suggestion
<sect2>Removal of the <em>cachemgr.cgi</em> tool.
```

File: doc/release-notes/release-7.sgml.in
Comment (rousskov) [inline doc/release-notes/release-7.sgml.in:53]: Technically, "it" refers to Cache Manager API or URL-path prefix in this context:

```suggestion
   instead of cachemgr.cgi.
```

File: doc/release-notes/release-7.sgml.in
Comment (rousskov) [inline doc/release-notes/release-7.sgml.in:47]: "these days" sounds like an overstatement -- I bet those tools were "more familiar" to (most) administrators for a very long time :-)

```suggestion
<p>Popular command-line tools such as <em>curl</em> or <em>wget</em> 
   provide equivalent features.
```

**Review by rousskov — APPROVED** [review]
I do not insist on any changes.

File: doc/release-notes/release-7.sgml.in
Comment (yadij) [reply doc/release-notes/release-7.sgml.in:49]: We do. An oddity of autotools on some OS.

**Review by rousskov — APPROVED** [review]

**Review by kinkie — APPROVED** [review]
ship it!

File: doc/release-notes/release-7.sgml.in
Comment (rousskov) [inline doc/release-notes/release-7.sgml.in:46]:
```
running: git -c core.whitespace=-blank-at-eof diff --check refs/pull/1542/merge^1
  doc/release-notes/release-7.sgml.in:46: trailing whitespace.
  +<p>Popular command-line tools such as <em>curl</em> or <em>wget</em> 
  Error:  git diff detected bad whitespace changes.
```

The suggestion below should address the above problem detected by [CI](https://github.com/squid-cache/squid/actions/runs/6646976944/job/18061486478?pr=1542).


```suggestion
<p>Popular command-line tools such as <em>curl</em> or <em>wget</em>
```

**Review by kinkie — APPROVED** [review]

**Review by rousskov — DISMISSED** [review]

**Review by kinkie — APPROVED** [review]

File: tools/Makefile.am
Comment (rousskov) [inline tools/Makefile.am:9]:
```suggestion
EXTRA_DIST = helper-ok-dying.pl helper-ok.pl

```

File: po4a.conf
Comment (rousskov) [inline po4a.conf:47]:
```suggestion
[type: man] src/squid.8.in $lang:doc/manuals/$lang/squid.8.in

```

**Review by rousskov — APPROVED** [review]
I left suggestions for two formatting fixes (only one of them was reported by CI but GitHub does flag both in PR code changes view). These fixes can be done later -- I do not insist on them.

**Comment by kinkie** [comment]
I'm worried that applying the changes will invalidate the approvals :\ I'm merging as-is to unblock other work

**Review by rousskov — COMMENTED** [review]
I am not sure what exactly went wrong because forced push erased breadcrumbs, but it looks like that recent force-push broke this PR. FWIW, please keep in mind that forced pushes to PR branches are very rarely _necessary_ and, in most other cases, create more problems than they solve.

**Comment by rousskov** [comment]
> I'm worried that applying the changes will invalidate the approvals

Yes, it would have invalidated the approvals, but, in this particular case, those invalidations would not cause any additional delays because this PR is authored by Amos (so his implicit approval does not get invalidated by changes), the changes would be committed by you (so you would, presumably, immediately re-approve), and the PR is more than two days old (so my/third approval would not speed things up). If it were your PR, I would have committed these trivial changes myself (instead of posting change requests) under to our "Easier Done" agreement, but I do not have a similar permission from Amos.

FWIW, this invalidation is a GitHub configuration option. We can turn it off, but I think it is the necessary evil, especially for PRs coming from folks that are not intimate with git (and may accidentally push unwanted changes that may get merged automatically). There are other solutions to mitigate that risk, but, AFAICT, they all require development and they all make the process more complex by adding more exceptions or special rules.

**Comment by yadij** [comment]
Had to rebase manually due to conflicts.

**Comment by rousskov** [comment]
> Had to rebase manually due to conflicts.

Resolving merge conflicts does not require rebasing.

**Review by rousskov — APPROVED** [review]
Approving without a careful review under the assumption that there were no material charges since the last review.

**Review by kinkie — APPROVED** [review]
Expediting along. I’m confident any issues would have been caught by CI

**Pull request closed without merging**
